beta-src/src/utils/map/writeBuildNotifications.ts

```diff
@@ -0,0 +1,87 @@
+import { current } from "@reduxjs/toolkit";
+import { GameState } from "../../state/interfaces/GameState";
+
+const defaultStyle = {
+  borderRadius: "8px",
+  backgroundColor: "rgba(0,0,0,.4)",
+  color: "#ffffff",
+  fontSize: "16px",
+  height: 40,
+  lineHeight: 2.5,
+  m: 1,
+  textAlign: "center",
+};
+
+// Currently, indexes 0 and 1 are being used as the delete labels.
+// 0 is the tracker which tells the player how many units need to be destroyed and is always displayed
+// when needing to destroy a unit and 1 renders conditionally based on whether the player has
+// selected any units so far or not
+
+/* eslint-disable no-param-reassign */
+export default function writeBuildNotifications(state): void {
+  const {
+    ordersMeta,
+    overview: { phase },
+  }: {
+    // definition
+    ordersMeta: GameState["ordersMeta"];
+    overview: { phase: GameState["overview"]["phase"] };
+  } = current(state);
+  if (phase === "Builds") {
+    let ordersToGo = 0;
+    let totalOrders = 0;
+    let toDo;
+    // values instead of entries key/value difference
+    Object.values(ordersMeta).forEach(({ update }) => {
+      if (update?.toTerrID === null) ordersToGo += 1;
+      totalOrders += 1;
+      if (update?.type === "Destroy") toDo = "Destroy";
+    });
+    if (toDo === "Destroy") {
+      if (ordersToGo > 1) {
+        state.notifications[0] = {
+          message: `Select ${ordersToGo} units to destroy.`,
+          style: {
+            ...defaultStyle,
+            fontWeight: 700,
+            width: 200,
+          },
+        };
+      } else if (ordersToGo === 1) {
+        state.notifications[0] = {
+          message: "Select 1 unit to destroy.",
+          style: {
+            ...defaultStyle,
+            fontWeight: 700,
+            width: 200,
+          },
+        };
+      } else if (ordersToGo === 0) {
+        state.notifications[0] = {
+          message: "Orders ready to submit",
+          style: {
+            ...defaultStyle,
+            fontWeight: 700,
+            width: 200,
+          },
+        };
+      }
+      if (ordersToGo < totalOrders) {
+        state.notifications[1] = {
+          message: "Select units again to cancel",
+          style: {
+            ...defaultStyle,
+            fontWeight: 400,
+            width: 218,
+          },
+        };
+      } else {
+        state.notifications[1] = undefined;
+      }
+    }
+  } else {
+    // If the current phase is not build, make sure the delete indexes are empty
+    state.notifications[0] = undefined;
+    state.notifications[1] = undefined;
```

Comment:
why not just reset the state notifications by setting them to an empty array?

Reply:
What I was thinking is that in the future, writeNotifications.ts is going to have a lot of specific phase notification writers so you want to make sure none of the indexes being used by other writers gets overwritten.

Reply:
Yes, but none of the phases should overlap, so you shouldn't get a situation where the dev needs to write diplomacy notifications and destroy notifications at the same time. 

If that were the case though, then we should split notifications into "phases" or "turn types" so that we store notifications there. 

For example: 

```
notifications: {
    "destroy": [],
    "build": [],
    ...
}
```